### PR TITLE
try to add authentication token for python script engine

### PR DESCRIPTION
**Pull request by sophiesongge**

### CODE REVIEW

**Comment by sophiesongge** [comment]
retest this please

**Comment by sophiesongge** [comment]
retest this please

**Comment by fviale** [comment]
you need to upgrade the script engine version in gradle.properties

**Review by fviale — APPROVED** [review]
LGTM

File: src/main/java/jsr223/cpython/PythonScriptEngine.java [hunk 1]
```diff
@@ -70,8 +71,11 @@
     public PythonScriptEngine() {
     }
 
-    private static synchronized GatewayServer startGatewayServer(EntryPoint entryPoint) {
-        GatewayServer gatewayServer = new GatewayServer(entryPoint, 0);
+    private static synchronized GatewayServer startGatewayServer(EntryPoint entryPoint, String authToken) {
+        GatewayServer gatewayServer = new GatewayServer.GatewayServerBuilder().entryPoint(entryPoint)
+                                                                              .authToken(authToken)
+                                                                              .javaPort(0)
```
Comment (MaelAudren) [inline src/main/java/jsr223/cpython/PythonScriptEngine.java:77]: what is the purpose of ```javaPort(0)``` ?

Comment (sophiesongge) [reply src/main/java/jsr223/cpython/PythonScriptEngine.java:77]: 0 is a reserved keyword in py4j, which means that the next available port will be used.

**Review by Mykhailenko — APPROVED** [review]

**Review by aminelouati — APPROVED** [review]

**Pull request merged**